spec/openapi.yaml

```diff
@@ -53,182 +156,498 @@ paths:
+              {
+                    'category': 'Tutorial',
+                    'name': 'CSS Tutorial',
+                    'paid': false,
+                    'url': 'https://www.w3schools.com/css/'
+                }
         required: true
       responses:
         200:
           description: Added new resource
-        404:
-          description: The specified resource was not found
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/Error'
+                $ref: '#/components/schemas/Resource'
+              example: |
+                    {
+                        'apiVersion': '1.0',
+                        'data': {
+                            'category': 'Tutorial',
+                            'created_at': '2019-07-21 15:41:19',
+                            'downvotes': 0,
+                            'id': 2137,
+                            'languages': [],
+                            'last_updated': '2019-07-22 16:05:53',
+                            'name': 'Some, but not ALL of the knowledge',
+                            'notes': null,
+                            'paid': false,
+                            'times_clicked': 3,
+                            'upvotes': 1,
+                            'url': 'http://www.test.com'
+                        },
+                        'status': 'ok',
+                        'status_code': 200
+                      }
+        401:
+          $ref: '#/components/responses/Unauthorized'
       security:
       - ApiKeyAuth: []
   /search:
     get:
-      description: Searches the API for a given keyword
-      responses:
-        200: {}
-  /apikey:
-    post:
-      description: Request API key by providing Operation Code credentials
+      summary: Search
+      description: Searches for resources containing a given keyword. For example, to search for any resource containing the word `algorithm`, make a `GET` request to `search?q=algorithm`
+      parameters:
+        - in: query
+          name: q
+          schema:
+            type: string
+          required: true
       responses:
         200:
-          description: API Key response
+          description: Search results returned
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/ApiKeyResponse'
-        401:
-          description: Unauthorized
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Error'
+                $ref: '#/components/schemas/SearchResponse'
+        404:
+          $ref: '#/components/responses/NotFound'
+  
 components:
   schemas:
     Error:
-      required:
-      - code
-      - message
       type: object
       properties:
-        code:
+        apiVersion: 
           type: string
-        message:
+        data: 
+          type: object
+        errors: 
           type: string
```

Comment:
`errors` is an array of objects, not a string. This screws up the example.

Reply:
Oh, and each object should contain a `code` and a `message`. I actually noticed there's one endpoint where this is not true, and I'm going to open an issue in the API to fix this. This has been great, thanks so much for doing all this!

Reply:
Oh, and validation errors will have `missing_params` and `invalid_params` (as soon as that PR is merged in the API anyway), so we'll want to make sure that the 422 reflects those as well. But 404 and the rest of the 4xx series should be fine with just `code` and `message`